spec/machine/ARM/MachineOps.thy

```diff
@@ -143,25 +143,29 @@ definition
 where
   "MAX_PERIOD_US \<equiv> 60 * 60 * 1000 * 1000"
 
-axiomatization
-  us_to_ticks :: "ticks \<Rightarrow> ticks"
-where
-  us_to_ticks_mono[intro!]: "mono us_to_ticks"
+definition "\<mu>s_in_ms = 1000"
+
+consts' factor1 :: ticks
+consts' factor2 :: ticks
+
+definition
+  "us_to_ticks us = (us * factor1) div factor2"
+
+axiomatization where
```

Comment:
Cool, the new axioms all look safe to me. Good job!

To really make sure, we should separately (in a test theory or a notepad or something where the axioms are not in scope) show that they are satisfiable. I.e. pick concrete numbers `kernelWCET_us`, `factor1`, `factor2`, etc and show that we can prove these rules. Ok to do that outside this PR, but it would be good to commit to the repo.

At some point later I'd like to auto-generate that for the kernel config, then we can replace the axioms, but that is in the future.

Reply:
Sure, I can have a go at that

Reply:
I've pushed a new commit that has a first attempt at this. I wasn't sure how to add a `defines` line for a function: when I tried, the "argument" to the function was appearing as an unknown schematic variable. So I've got some lemma statements that are like an unfolding of the `us_to_ticks` mentioned in the axioms.

(I haven't made any other changes for the PR yet, but I'll get to that soon.)

I'm not sure where it would be best to have this. It's not dependent on the architecture, so it shouldn't be in `MachineOps.thy`. Maybe a separate file immediately after?

Reply:
Well, in saying that it isn't dependent on the architecture, `MAX_PERIOD_US` comes from the C config file, so I guess that's some sort of dependence, and we'd want to be using that number in this demonstration...

Reply:
The two factors will eventually also come out of the kernel config and will depend on the platform, so I think that is Ok to leave in here.